src/pages/import/helpers.js

```diff
@@ -89,121 +118,154 @@ function isFetchLogComplete(data) {
                 eventEmitter.emit('summary.totals', data.stats)
                 eventEmitter.emit('summary.typeReports', data.typeReports)
             } else if (data.conflicts) {
+                debugFetchTaskSummary(`Fetch task summary, has "conflicts"`)
                 eventEmitter.emit('summary.importCount', data.importCount)
                 eventEmitter.emit('summary.conflicts', data.conflicts)
             } else if (data.importSummaries) {
+                debugFetchTaskSummary(`Fetch task summary, has "importsSummaries"`)
                 eventEmitter.emit('summary.importCount', data)
                 eventEmitter.emit(
                     'summary.importSummaries',
                     data.importSummaries
                 )
             } else if (data.importCount) {
+                debugFetchTaskSummary(`Fetch task summary, has "importCount"`)
                 eventEmitter.emit('summary.importCount', data.importCount)
             } else {
-                console.error(
-                    'No summary generated. Receieved data not recognized: ',
-                    data
-                )
+                debugFetchTaskSummary(`Fetch task summary, error, data: %O`, data)
             }
         }
 
-        eventEmitter.emit('summary.loaded')
+        throw new Error(`No summary could be loaded, data is: ${data}`)
     } catch (e) {
-        !isProduction && console.log(`Task Summaries: Error fetching ${type}`)
-        !isProduction && console.log(e)
+        debugFetchTaskSummary(`Fetch task summary, fetching error for %s(%s): %s`, jobId, type, e.message)
     }
+
+    eventEmitter.emit('summary.loaded')
 }
 
+const debugLogStats = debugLog.extend('stats')
 function logStats(stats, type) {
     if (!stats) {
+        debugLogStats(`Log stats, no data received`)
         return
     }
 
     const { created, updated, deleted, ignored, total } = stats
-    eventEmitter.emit('log', {
+    const logData = {
         id: new Date().getTime(),
         d: null,
         type: `INFO - ${CATEGORY_2_LABEL[type]}`,
         text: `Created: ${created}, Updated: ${updated}, Deleted: ${deleted}, Ignored: ${ignored}, Total: ${total}`,
-    })
+    }
+
+    debugLogStats(`Log stats: %O`, logData)
+    eventEmitter.emit('log', logData)
 }
 
+const debugLogCount = debugLog.extend('stats')
 function logImportCount(importCount, type) {
     if (!importCount) {
+        debugLogCount(`Log import count, no data received`)
         return
     }
 
     const { deleted, ignored, imported, updated } = importCount
-    eventEmitter.emit('log', {
+    const logData = {
         id: new Date().getTime(),
         d: new Date(),
         type: `INFO - ${CATEGORY_2_LABEL[type]}`,
         text: `Imported: ${imported}, Updated: ${updated}, Deleted: ${deleted}, Ignored: ${ignored}`,
-    })
+    }
+
+    debugLogCount(`Log import count: %O`, logData)
+    eventEmitter.emit('log', logData)
 }
 
+const debugLogConflicts = debugLog.extend('stats')
```

Comment:
Not sure about all these `extend`:ed loggers in the same file. It's a bit messy.

Reply:
At first, I tried it without extending inside the file, which makes the file/debug code look cleaner.
But unlike on a cli, objects won't be printed as a string in the browser's console, so having all the debug call's namespaces printed in the same color makes it very hard to get a "quick" overview what ran in which order, which is why I added all the `extend`s.

I wouldn't mind reverting this, but I suggest you try both versions yourself as well.
(For cli scripts I def. wouldn't have done this)

Reply:
Agreed. To me this is a signal that maybe `debug` isn't the right tool for the job.